src/sincere_singularities/modules/restaurants_view.py

```diff
@@ -63,24 +65,33 @@ def __init__(self, restaurants: "Restaurants", embeds: list[disnake.Embed], inde
         for i, embed in enumerate(self.embeds):
             embed.set_footer(text=f"Restaurant {i + 1} of {len(self.embeds)}")
 
-        self._update_state()
+        self.update_state()
 
-    def _update_state(self) -> None:
+    def update_state(self) -> None:
+        """Updating the State of the RestaurantsView"""
         # Disable previous/next button for first/last embeds
         self._prev_page.disabled = self.index == 0
         self._next_page.disabled = self.index == len(self.embeds) - 1
+        if has_restaurant(self.restaurants.interaction.user.id, self.restaurants.all_restaurants[self.index].name):
+            self._enter_restaurant.label = "Enter restaurant"
+        else:
+            self._enter_restaurant.label = "Buy"
+        points = get_points(self.restaurants.interaction.user.id)
+        description = self.embeds[self.index].description
+        assert description
+        self.embeds[self.index].description = re.sub(r"you have \d+", f"you have {points}", description)
```

Comment:
btw doesn't the "You (don't) own this restaurant" text also needs updating?

Reply:
Fixed by "proxying" the embed property of RestaurantView to Restaurants, which we should have probably done anyways.